bitnami/harbor/README.md

```diff
@@ -0,0 +1,274 @@
+
+```bash
+cd charts/bitnami/harbor
+```
+
+### Configure the chart
+
+The following items can be configured in `values.yaml` or set via `--set` flag during installation.
+
+#### Configure the way how to expose Harbor service:
+
+- **Ingress**: The ingress controller must be installed in the Kubernetes cluster.
+  **Notes:** if the TLS is disabled, the port must be included in the command when pulling/pushing images. Refer to issue [#5291](https://github.com/goharbor/harbor/issues/5291) for the detail.
+- **ClusterIP**: Exposes the service on a cluster-internal IP. Choosing this value makes the service only reachable from within the cluster.
+- **NodePort**: Exposes the service on each Node’s IP at a static port (the NodePort). You’ll be able to contact the NodePort service, from outside the cluster, by requesting `NodeIP:NodePort`.
+- **LoadBalancer**: Exposes the service externally using a cloud provider’s load balancer.
+
+#### Configure the external URL
+
+The external URL for Harbor core service is used to:
+
+1. populate the docker/helm commands showed on portal
+2. populate the token service URL returned to docker/notary client
+
+Format: `protocol://domain[:port]`. Usually:
+
+- if expose the service via `Ingress`, the `domain` should be the value of `service.ingress.hosts.core`
+- if expose the service via `ClusterIP`, the `domain` should be the value of `service.clusterIP.name`
+- if expose the service via `NodePort`, the `domain` should be the IP address of one Kubernetes node
+- if expose the service via `LoadBalancer`, set the `domain` as your own domain name and add a CNAME record to map the domain name to the one you got from the cloud provider
+
+If Harbor is deployed behind the proxy, set it as the URL of proxy.
+
+#### Configure data persistence:
+
+- **Disable**: The data does not survive the termination of a pod.
+- **Persistent Volume Claim(default)**: A default `StorageClass` is needed in the Kubernetes cluster to dynamically provision the volumes. Specify another StorageClass in the `storageClass` or set `existingClaim` if you have already existing persistent volumes to use.
+- **External Storage(only for images and charts)**: For images and charts, the external storages are supported: `azure`, `gcs`, `s3` `swift` and `oss`.
+
+#### Configure the secrets
+
+- **Secret keys**: Secret keys are used for secure communication between components. Fill `core.secret`, `jobservice.secret` and `registry.secret` to configure.
+- **Certificates**: Used for token encryption/decryption. Fill `core.secretName` to configure.
+
+Secrets and certificates must be setup to avoid changes on every Helm upgrade (see: [#107](https://github.com/goharbor/harbor-helm/issues/107)).
+
+#### Configure the other items listed in [configuration](#configuration) section.
+
+### Install the chart
+
+Install the Harbor helm chart with a release name `my-release`:
+
+```bash
+helm install --name my-release --set service.tls.commonName=your.comain.com .
+```
```

Comment:
I am still unable to access the portal from outside. So, put some instructions here on how to access it

Reply:
After executing this install command you get the following info from the NOTES.txt:

```
** Please be patient while the chart is being deployed **

1. Get the Harbor URL:

  echo "Harbor URL: https://127.0.0.1:8443/"
  kubectl port-forward --namespace tomas svc/harbor 8443:443

2. Login with the following credentials to see your Harbor application

  echo Username: "admin"
  echo Password: $(kubectl get secret --namespace tomas my-release-harbor-core -o jsonpath="{.data.HARBOR_ADMIN_PASSWORD}" | base64 --decode)
```

Do you have issues with those instructions?